Portal/src/Datahub.Portal/Pages/Workspace/Storage/Heading.razor

```diff
@@ -128,6 +131,37 @@
         var currentUser = await _userInformationService.GetCurrentPortalUserAsync();
         var projectUsers = await _projectUserManagementService.GetProjectUsersAsync(ProjectAcronym);
         _currentUserRole = projectUsers.FirstOrDefault(pu => pu.PortalUser.Id == currentUser.Id)?.Role;
+        
+        await CheckPublishingBlocklistAsync();
+    }
+
+    private async Task CheckPublishingBlocklistAsync()
+    {
+        try
+        {
+            // Get the workspace lead
+            var workspaceLead = await _projectUserManagementService.GetProjectLeadAsync(ProjectAcronym);
+            
+            if (workspaceLead?.PortalUser?.Email != null)
```

Comment:
This section of code is repeated in multiple places. Why isn't it part of the block service? Also can we cache the result to avoid all the repeated db calls